packages/plugin-ext/src/main/browser/debug/debug-main.ts

```diff
@@ -114,11 +115,17 @@ export class DebugMainImpl implements DebugMain, Disposable {
     }
 
     async $appendToDebugConsole(value: string): Promise<void> {
-        this.debugConsoleSession.append(value);
+        const session = this.consoleSessionManager.selectedSession;
+        if (session instanceof DebugConsoleSession) {
+            session.append(value);
+        }
     }
```

Comment:
A quick note on that: I'm wondering whether this makes sense at all. Do we somehow know which extension a debug console belongs to? Currently, it will just log any input to the currently selected console session, but I'm pretty sure that's not the expected behavior. @vince-fugnitto WDYT?

Reply:
I'm not sure if we have the information or can deduce it. I believe on `master` before your changes all output was displayed in the same console. Your changes are more aligned with the behavior I notice in vscode.

Reply:
Ha, I wanted to see how vscode does it. I think I accidentally replicated [their code](
https://github.com/microsoft/vscode/blob/e657d42281487040b50374d482d34b3fb05eb377/src/vs/workbench/api/browser/mainThreadDebugService.ts#L281-L287). It seems like vscode also just logs to the selected session if `append` or `appendLine` calls are made. So I guess that resolves that question.

I also couldn't find any vscode extension that actually calls these methods.